domain/access/state/permission.go

```diff
@@ -644,6 +648,86 @@ AND    m.cloud_credential_name = $input.cred_name
 	return results, nil
 }
 
+// ImportOfferAccess imports the user access for offers in the model.
+func (st *PermissionState) ImportOfferAccess(
+	ctx context.Context,
+	importAccess []access.OfferImportAccess,
+) error {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	userNames := set.NewStrings()
+	for _, imp := range importAccess {
+		userNames = userNames.Union(set.NewStrings(slices.Collect(maps.Keys(imp.Access))...))
+	}
+	userNames.Add(corepermission.EveryoneUserName.String())
```

Comment:
Neat. I like it.

If you wanted to do it without a string set, you could do this:
```suggestion
	userNames := []string{
		corepermission.EveryoneUserName.String(),
	}
	for _, imp := range importAccess {
		userNames = slices.AppendSeq(userNames, maps.Keys(imp.Access))
	}
	userNames = slices.Compact(slices.Sort(userNames))
```